docs/dev/allocation.qmd

```diff
@@ -1,7 +1,7 @@
 ---
 title: "Allocation"
 ---
-# Overview of allocation implementation
+# Overview of allocation implementation [#sec-allocation-overview]
```

Comment:
![image](https://github.com/user-attachments/assets/0aaacb1c-478d-4fbd-ae1e-3224f10e2bb8)
After rendering, this section looks like this. It doesn't seem correct.